consensus/hotstuff/eventloop/event_loop_v2.go

```diff
@@ -1,53 +1,68 @@
 package eventloop
 
 import (
+	"context"
 	"time"
 
 	"github.com/rs/zerolog"
 
 	"github.com/onflow/flow-go/consensus/hotstuff"
 	"github.com/onflow/flow-go/consensus/hotstuff/model"
-	"github.com/onflow/flow-go/engine"
 	"github.com/onflow/flow-go/model/flow"
 	"github.com/onflow/flow-go/module"
-	"github.com/onflow/flow-go/module/lifecycle"
+	"github.com/onflow/flow-go/module/component"
+	"github.com/onflow/flow-go/module/irrecoverable"
 	"github.com/onflow/flow-go/module/metrics"
 )
 
 // EventLoopV2 buffers all incoming events to the hotstuff EventHandler, and feeds EventHandler one event at a time.
 type EventLoopV2 struct {
+	*component.ComponentManager
 	log                zerolog.Logger
 	eventHandler       hotstuff.EventHandlerV2
 	metrics            module.HotstuffMetrics
 	proposals          chan *model.Proposal
 	quorumCertificates chan *flow.QuorumCertificate
-
-	lm   *lifecycle.LifecycleManager
-	unit *engine.Unit // lock for preventing concurrent state transitions
+	startTime          time.Time
 }
 
 var _ hotstuff.EventLoopV2 = &EventLoopV2{}
 var _ module.ReadyDoneAware = &EventLoopV2{}
+var _ module.Startable = &EventLoopV2{}
 
 // NewEventLoopV2 creates an instance of EventLoopV2.
-func NewEventLoopV2(log zerolog.Logger, metrics module.HotstuffMetrics, eventHandler hotstuff.EventHandlerV2) (*EventLoopV2, error) {
+func NewEventLoopV2(log zerolog.Logger, metrics module.HotstuffMetrics, eventHandler hotstuff.EventHandlerV2, startTime time.Time) (*EventLoopV2, error) {
 	proposals := make(chan *model.Proposal)
 	quorumCertificates := make(chan *flow.QuorumCertificate)
 
 	el := &EventLoopV2{
 		log:                log,
-		lm:                 lifecycle.NewLifecycleManager(),
 		eventHandler:       eventHandler,
 		metrics:            metrics,
 		proposals:          proposals,
 		quorumCertificates: quorumCertificates,
-		unit:               engine.NewUnit(),
+		startTime:          startTime,
 	}
 
+	componentBuilder := component.NewComponentManagerBuilder()
+	componentBuilder.AddWorker(func(ctx irrecoverable.SignalerContext, ready component.ReadyFunc) {
+		ready()
+
+		// launch when scheduled by el.startTime
```

Comment:
better to add some info logs saying, `event loop will start at XXX` and `event loop has started`